tensorflow_datasets/translate/opus.py

```diff
@@ -0,0 +1,266 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""opus dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from absl import logging
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@inproceedings{Tiedemann2012ParallelData,
+  author = {Tiedemann, J},
+  title = {Parallel Data, Tools and Interfaces in OPUS},
+  booktitle = {LREC}
+  year = {2012}}
+"""
+
+_DESCRIPTION = """
+OPUS is a collection of translated texts from the web.
+
+Create your own config to choose which data / language pair to load.
+
+```
+config = tfds.translate.opus.OpusConfig(
+    version=tfds.core.Version('0.1.0'),
+    language_pair=("de", "en"),
+    subsets=["GNOME", "EMEA"]
+)
+builder = tfds.builder("opus", config=config)
+```
+"""
+
+class SubDataset():
+  """Class to keep track of information on a sub-dataset of OPUS."""
+
+  def __init__(self, name, description, homepage, url, languages, filename):
+    """Sub-dataset of OPUS.
+
+    Args:
+      name: `string`, a unique dataset identifier.
+      description: `string`, a description of the dataset.
+      homepage: `string`, homepage of the dataset.
+      url: `string`, download url for the dataset.
+      languages: `<list>(string)`, a list of supported languages.
+      filename: `string`, the prefix of the downloaded files.
+    """
+    self.name = name
+    self.description = description
+    self.homepage = homepage
+    self.url = url
+    self.filename = filename
```

Comment:
`name` and `filename` both same? I suspect duplicate parameters. You could use anyone instead of two.

Reply:
Done.